### DISCUSSION TITLE
updateTriggers does not work with MVTLayer and sub layers

### DISCUSSION THREAD

**Discussion by hifichevymane** [visgl/deck.gl#8689] [Q&A] [answered]
Hey, I have a web app built on `MapboxGL` and `deck.gl`. I use `MVTLayer` to render my points on the map and pass in `renderSubLayers` `GeoJsonLayer` with custom options. My `GeoJsonLayer` render points on the map as icons. I want to implement selecting points on click so when I click on the point the icon should change, I've implemented it using `updateTriggers` and `onClick`. I return `feature_id` from the backend to get a unique id of the point. I store this value in `properties` property of `info.object` object

But when I click on the point it does not change, it starts to change when I zoom out or zoom in. The points also disappear when I drag the map. I cannot understand where is the problem. Can you help me figure out what can be the problem? 

Here's the getters that return my `MVTLayer` with point icon  **(I use JavaScript custom HTML elements approach to build components. I pass `this._selectedPoint = null;` in the constructor to keep track of selected point)**:
```
  get pointsLayer () {
    return new MVTLayer({
      id: 'locations-markers-points',
      data: `api/points?zoom={z}&x={x}&y={y}`,
      uniqueIdProperty: 'feature_id',
      parameters: {
        depthTest: false
      },
      // Interaction settings
      pickable: true,
      renderSubLayers: props => {
        return new GeoJsonLayer(props, {
          pointType: 'icon',
          getIcon: d => ({
            url: this.getPointIcon(d.properties.feature_id),
            x: 0,
            y: 0,
            width: 17,
            height: 17
          }),
          getIconSize: 10,
          iconSizeScale: 2,
          pickable: true,
          updateTriggers: {
            getIcon: [this._selectedPoint]
          }
        });
      },
      onClick: ({ object }) => {
        this._selectedPoint = { ...object };
        console.log(this._selectedPoint);
      }
    });
  }
  
  getPointIcon (currentPointFeatureId) {
    if (this._selectedPoint &&
        this._selectedPoint.properties.feature_id === currentPointFeatureId) {
      return 'Map/img/selected-square-point-icon.svg';
    }
    return 'Map/img/square-point-icon.svg';
  }
```

https://github.com/visgl/deck.gl/assets/106327285/b597c08c-ff5a-43a2-b0f2-baca3e806264

**UPD:** `updateTriggers` property just don't work when I change `this._selectedPoint`

**Comment by Pessimistress** [comment, marked as answer]
`renderSubLayers` result is cached once a tile has been rendered. You need to set `updateTriggers` of the MVTLayer to trigger an update.

Also you can directly pass any GeoJsonLayer props to the MVTLayer: https://deck.gl/docs/api-reference/geo-layers/mvt-layer#properties

**Reply by hifichevymane** [reply]
Thank you, I have another question. If `updateTriggers` is triggered, will deck.gl rerender this layer or do we have to do it by ourselves?
